File: Arrays/53-Maximum-Subarray/JavaScript/Python/53-Maximum-Subarray-Brute Force.py
"""
Problem: 53. Maximum Subarray
Approach: Brute Force

Time Complexity: O(n^2)
Space Complexity: O(1)
"""


class Solution:
    def maxSubArray(self, nums: list[int]) -> int:
        maxi = float("-inf")

        for i in range(len(nums)):
            total = 0

            for j in range(i, len(nums)):
                total += nums[j]
                maxi = max(maxi, total)

        return maxi

File: Arrays/53-Maximum-Subarray/JavaScript/53-Maximum-Subarray-Brute Force.js
/**
 * Problem: 53. Maximum Subarray
 * Approach: Brute Force
 *
 * Time Complexity: O(n^2)
 * Space Complexity: O(1)
 */

/**
 * @param {number[]} nums
 * @return {number}
 */

var maxSubArray = function(nums) {
    let maxi = -Infinity;

    for (let i = 0; i < nums.length; i++) {
        let total = 0;

        for (let j = i; j < nums.length; j++) {
            total += nums[j];
            maxi = Math.max(maxi, total);
        }
    }

    return maxi;
};